packages/fx-core/src/plugins/resource/aad/aadAppClient.ts

```diff
@@ -105,9 +106,13 @@ export class AadAppClient {
     skip = false
   ): Promise<void> {
     try {
-      await AadAppClient.retryHanlder(stage, () =>
-        AadAppManifestManager.updateAadApp(TokenProvider.token as string, manifest)
-      );
+      await AadAppClient.retryHanlder(stage, async () => {
+        const preAuthorizedApplications = manifest.preAuthorizedApplications;
+        manifest.preAuthorizedApplications = [];
+        await AadAppManifestManager.updateAadApp(TokenProvider.token as string, manifest);
+        manifest.preAuthorizedApplications = preAuthorizedApplications;
+        await AadAppManifestManager.updateAadApp(TokenProvider.token as string, manifest);
```

Comment:
will it fail, if there is already pre-authed apps?

Reply:
We first only update oauth2Permissions and set pre-auth apps with empty value, it will not fail. 
pre-auth apps should be update after oauth2permissions